### ISSUE TITLE
Launch crash with Optifine

### ISSUE THREAD

**Issue by Madis0** [Siphalor/amecs-api#1]
Initially tested with

- Minecraft 1.14.3
- Amecs 1.1.0
- OptiFine 1.14.3 HD U F1
- OptiFabric 0.3.1
- Fabric Loader 0.4.8 build.155 1.14.3
- Fabric API 0.3.0 build.187

Still reproducable with

- Minecraft 1.14.4
- Amecs 1.1.0
- OptiFine 1.14.4 HD U F2 pre1
- OptiFabric 0.4.3
- Fabric Loader 0.4.8 build.155 1.14.4
- Fabric API 0.3.0 build 200

```
---- Minecraft Crash Report ----
// But it works on my machine.

Time: 30.06.19 16:20
Description: Initializing game

org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:521)
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.loadClassData(KnotClassDelegate.java:179)
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:142)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at net.minecraft.class_310.method_1503(class_310.java:441)
	at net.minecraft.class_310.method_1514(class_310.java:395)
	at net.minecraft.client.main.Main.main(Main.java:154)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:170)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:129)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Callback method handler$onKeyBindingWritten$zzf000(Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;Ljava/io/PrintWriter;[Lnet/minecraft/class_304;IILnet/minecraft/class_304;)V in amecs.mixins.json:MixinGameOptions failed injection check, (0/1) succeeded. Using refmap amecs-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:290)
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1203)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:939)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:323)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:281)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:353)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.apply(MixinTransformer.java:724)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.applyMixins(MixinTransformer.java:703)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:509)
	... 13 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:521)
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.loadClassData(KnotClassDelegate.java:179)
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:142)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at net.minecraft.class_310.method_1503(class_310.java:441)

-- Initialization --
Details:
Stacktrace:
	at net.minecraft.class_310.method_1514(class_310.java:395)
	at net.minecraft.client.main.Main.main(Main.java:154)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:170)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:129)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)

-- System Details --
Details:
	Minecraft Version: 1.14.3
	Operating System: Linux (amd64) version 4.19.49-1-MANJARO
	Java Version: 1.8.0_212, Oracle Corporation
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 926272976 bytes (883 MB) / 1140850688 bytes (1088 MB) up to 2147483648 bytes (2048 MB)
	JVM Flags: 7 total; -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Launched Version: fabric-loader-0.4.8+build.155-1.14.3
	LWJGL: 3.2.2 build 10
	OpenGL: NO CONTEXT
	GL Caps: 
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	Resource Packs: ~~ERROR~~ MixinTransformerError: An unexpected critical error was encountered
	Current Language: ~~ERROR~~ NullPointerException: null
	CPU: <unknown>
```

**Comment by Siphalor** [comment]
No promises but this issue might be more permanently fixed in release 1.1.2 of Amecs API.
This version will probably arrive in Mouse Wheelie 1.6.4 and Amecs 1.3.3.
I hope that somebody can confirm this issue to be gone then.

**Comment by Siphalor** [comment]
As there were no more reports I just declare this thing fixed.
Please respond if this reoccurs.
